### ISSUE TITLE
0.7.1 install problem

### ISSUE THREAD

**Issue by nicutor** [sous-chefs/vagrant-cookbook#73]
Hi,

Trying to use vagrant cookbook 0.7.1 and I get this error.

* Windows 7 SP1 with latest updates
* chefdk 2.0.28
* PowerShell 5.1
* Microsoft Visual C++ 2010 x64 and x86

```
Generated at 2017-08-04 20:13:47 +0300
NoMethodError: undefined method `split' for nil:NilClass
c:/tmp/local-mode-cache/cache/cookbooks/vagrant/libraries/helpers.rb:66:in `extract_checksum'
c:/tmp/local-mode-cache/cache/cookbooks/vagrant/libraries/helpers.rb:31:in `vagrant_sha256sum'
c:/tmp/local-mode-cache/cache/cookbooks/vagrant/recipes/windows.rb:19:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `instance_eval'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/cookbook_version.rb:205:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:343:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:299:in `block in include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `each'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/tmp/local-mode-cache/cache/cookbooks/vagrant/recipes/default.rb:18:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `instance_eval'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/cookbook_version.rb:205:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:343:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:299:in `block in include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `each'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/tmp/local-mode-cache/cache/cookbooks/tmp/recipes/vagrant.rb:1:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `instance_eval'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/cookbook_version.rb:205:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:343:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:299:in `block in include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `each'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/tmp/local-mode-cache/cache/cookbooks/tmp/recipes/windows.rb:12:in `block in from_file'
c:/tmp/local-mode-cache/cache/cookbooks/tmp/recipes/windows.rb:11:in `each'
c:/tmp/local-mode-cache/cache/cookbooks/tmp/recipes/windows.rb:11:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `instance_eval'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/cookbook_version.rb:205:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:343:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:299:in `block in include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `each'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:298:in `include_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/tmp/local-mode-cache/cache/cookbooks/tmp/recipes/default.rb:7:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `instance_eval'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/mixin/from_file.rb:30:in `from_file'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/cookbook_version.rb:205:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:343:in `load_recipe'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:163:in `block in compile_recipes'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:160:in `each'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:160:in `compile_recipes'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:77:in `compile'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/run_context.rb:192:in `load'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/policy_builder/expand_node_object.rb:97:in `setup_run_context'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/client.rb:513:in `setup_run_context'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/client.rb:281:in `run'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application.rb:273:in `run_with_graceful_exit_option'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application.rb:249:in `block in run_chef_client'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/local_mode.rb:44:in `with_server_connectivity'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application.rb:232:in `run_chef_client'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application/client.rb:469:in `sleep_then_run_chef_client'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application/client.rb:458:in `block in interval_run_chef_client'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application/client.rb:457:in `loop'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application/client.rb:457:in `interval_run_chef_client'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application/client.rb:441:in `run_application'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/lib/chef/application.rb:59:in `run'
C:/opscode/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20-universal-mingw32/bin/chef-client:26:in `<top (required)>'
C:/opscode/chefdk/bin/chef-client:275:in `load'
C:/opscode/chefdk/bin/chef-client:275:in `<main>'
```

**Comment by jqassar** [comment]
Possible issue: The extension for the Windows download changed from vagrant_(version).msi to vagrant_(version)_(arch).msi as of Vagrant 1.9.6.

As a workaround prior to architecture flags being added, you might be able to either use the node attributes node['vagrant']['url'], node['vagrant']['checksum'], and node['vagrant']['msi_version'] to construct a literal path to the installation, or back down to a Vagrant version <= 1.9.5.

**Comment by MarkGibbons** [comment]
Should be fixed once #80 is published
